WalletWasabi/WabiSabi/Client/CoinJoinClient.cs

```diff
@@ -77,57 +86,67 @@ public async Task<bool> StartCoinJoinAsync(IEnumerable<SmartCoin> coins, Cancell
+				DependencyGraphTaskScheduler scheduler = new(dependencyGraph);
 
-			// Output registration.
-			roundState = await RoundStatusUpdater.CreateRoundAwaiter(roundState.Id, rs => rs.Phase == Phase.OutputRegistration, cancellationToken).ConfigureAwait(false);
-			await scheduler.StartOutputRegistrationsAsync(outputTxOuts, bobClient, cancellationToken).ConfigureAwait(false);
+				// Re-issuances.
+				var bobClient = CreateBobClient(roundState);
+				await scheduler.StartReissuancesAsync(registeredAliceClients, bobClient, cancellationToken).ConfigureAwait(false);
 
-			// ReadyToSign.
-			await ReadyToSignAsync(registeredAliceClients, cancellationToken).ConfigureAwait(false);
+				// Output registration.
+				roundState = await RoundStatusUpdater.CreateRoundAwaiter(roundState.Id, rs => rs.Phase == Phase.OutputRegistration, cancellationToken).ConfigureAwait(false);
+				await scheduler.StartOutputRegistrationsAsync(outputTxOuts, bobClient, cancellationToken).ConfigureAwait(false);
 
-			// Signing.
-			roundState = await RoundStatusUpdater.CreateRoundAwaiter(roundState.Id, rs => rs.Phase == Phase.TransactionSigning, cancellationToken).ConfigureAwait(false);
-			var signingState = roundState.Assert<SigningState>();
-			var unsignedCoinJoin = signingState.CreateUnsignedTransaction();
+				// ReadyToSign.
+				await ReadyToSignAsync(registeredAliceClients, cancellationToken).ConfigureAwait(false);
 
-			// Sanity check.
-			if (!SanityCheck(outputTxOuts, unsignedCoinJoin))
-			{
-				throw new InvalidOperationException($"Round ({roundState.Id}): My output is missing.");
-			}
+				// Signing.
+				roundState = await RoundStatusUpdater.CreateRoundAwaiter(roundState.Id, rs => rs.Phase == Phase.TransactionSigning, cancellationToken).ConfigureAwait(false);
+				var signingState = roundState.Assert<SigningState>();
+				var unsignedCoinJoin = signingState.CreateUnsignedTransaction();
 
-			// Send signature.
-			await SignTransactionAsync(registeredAliceClients, unsignedCoinJoin, cancellationToken).ConfigureAwait(false);
+				// Sanity check.
+				if (!SanityCheck(outputTxOuts, unsignedCoinJoin))
+				{
+					throw new InvalidOperationException($"Round ({roundState.Id}): My output is missing.");
+				}
+
+				// Send signature.
+				await SignTransactionAsync(registeredAliceClients, unsignedCoinJoin, cancellationToken).ConfigureAwait(false);
 
-			var finalRoundState = await RoundStatusUpdater.CreateRoundAwaiter(s => s.Id == roundState.Id && s.Phase == Phase.Ended, cancellationToken).ConfigureAwait(false);
-			return finalRoundState.WasTransactionBroadcast;
+				var finalRoundState = await RoundStatusUpdater.CreateRoundAwaiter(s => s.Id == roundState.Id && s.Phase == Phase.Ended, cancellationToken).ConfigureAwait(false);
+				return finalRoundState.WasTransactionBroadcast;
+			}
+			finally
+			{
+				State = CoinJoinClientState.InProgress;
+			}
```

Comment:
I think the try-finally is unnecessary because:

* in case there is an exception the coinjoin task finishes as faulted. (completed)
* in case everything goes right the coinjoin task finishes as successfully completed. (completed)
* in case everything goes right (but needs a blame round) the method is called again and the `State` is set to `InProgress` again at the begining.

Reply:
Sorry I did not meant to edit this - just quote answer.